app/controllers/placements/schools/placements/build_controller.rb

```diff
@@ -153,12 +159,22 @@ def build_placement
     end
   end
 
+  def setup_skipped_steps
+    return if school.mentors.exists?
+
+    session[:add_a_placement][:skipped_steps] = %w[add_mentors]
+  end
+
+  def skipped_steps
+    session.dig(:add_a_placement, "skipped_steps") || []
+  end
```

Comment:
There's absolutely nothing wrong with this approach, so no need to change it. But I do wonder whether we really need to store this in the session, or if it could be calculated dynamically.

Perhaps if/when we refactor this in the future, we could keep the 'conditional' bit closer to the step definition itself. For example (pseudo-code!):

```ruby
STEPS = [
  { name: :choose_phase, unless: -> { school.primary_or_secondary_only? } },
  { name: :choose_subject },
  { name: :choose_mentors, unless: -> { school.mentors.none? } },
  { name: :check_your_answers },
]
```

As we've said before, I think there's probably an opportunity to refine this a bit once the dust has settled a bit – or once we have more wizard-style flows to implement.

Reply:
This is kind of what I wanted to achieve but didn't know that you could do it this way, this is really interesting! Thanks for raising.

Reply:
Added to the refactor card: https://trello.com/c/HhyaTg8c/98-tidy-up-the-add-a-placement-multi-step-form-implementation